src/main/java/pl/sdacademy/carrental/services/ReservationService.java

```diff
@@ -0,0 +1,101 @@
+package pl.sdacademy.carrental.services;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import pl.sdacademy.carrental.domain.Branch;
+import pl.sdacademy.carrental.domain.cars.Car;
+import pl.sdacademy.carrental.domain.cars.CarCategory;
+import pl.sdacademy.carrental.mappers.CarToReservationListItemDetailsMapper;
+import pl.sdacademy.carrental.model.ReservationListDetails;
+import pl.sdacademy.carrental.repositories.ReservationRepository;
+import pl.sdacademy.carrental.requests.CarReservationRequest;
+
+import java.time.LocalDate;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Service
+@Transactional
+public class ReservationService {
+   
+   private final ReservationRepository reservationRepo;
+   private final BranchService branchService;
+   private final CarService carService;
+   private final CarToReservationListItemDetailsMapper carToReservationListItemMapper;
+   
+   public ReservationService(final ReservationRepository reservationRepo,
+                             final BranchService branchService,
+                             final CarService carService,
+                             final CarToReservationListItemDetailsMapper carToReservationListItemMapper) {
+      this.reservationRepo = reservationRepo;
+      this.branchService = branchService;
+      this.carService = carService;
+      this.carToReservationListItemMapper = carToReservationListItemMapper;
+   }
+   
+   public List<ReservationListDetails> getCarsUpForReservation(final CarReservationRequest request) {
+      final Branch pickupBranch = branchService.findBranchByName(request.getPickupBranchName());
+      final LocalDate pickupDate = LocalDate.parse(request.getPickupDate());
+      final LocalDate returnDate = LocalDate.parse(request.getReturnDate());
+      
+      Map<CarCategory, Integer> numberOfAvailableCarsPerCategoryInPeriod = new HashMap<>();
+      
+      Arrays.asList(CarCategory.values()).forEach(category -> {
```

Comment:
Stream.of(CarCategory.values()).forEach() and move lambda inside to a separate method

Reply:
Done, although I'm not sure about benefits of this change (`Stream.of(…)forEach(…)` over `Arrays.asList(…).forEach(…)`)